pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKSessionWatcher.java

```diff
@@ -81,7 +83,8 @@ public void close() throws Exception {
     }
 
     // task that runs every TICK_TIME to check zk connection
-    private synchronized void checkConnectionStatus() {
+    @VisibleForTesting
+    private void checkConnectionStatus() {
```

Comment:
if zk client can't ensure the order. We should add a comment to explain this method is not thread-safe. because we call it in the single thread, it will be fine. but we shouldn't leave any potential problems in the future.